domain/status/status.go

```diff
@@ -171,7 +171,7 @@ func RelationStatusTransitionValid(current, new StatusInfo[RelationStatusType])
 			validTransition = current.Status != RelationStatusTypeBroken
 		case RelationStatusTypeError:
 			if new.Message == "" {
-				return errors.Errorf("cannot set status %q without info", new.Status)
+				return errors.Errorf("cannot set status %q without message", new.Status)
```

Comment:
I realise that this PR is not the PR that has introduced this concept but I am worried this was written without some thought behind it.

Are we sure that this is smart? i.e this check should be happening at the service layer and not anywhere else. This check is about state machine transitions not that the user entered the data correctly. Also if this triggers in a production case it means we will never set the thing to an error state and instead leave it a state that implies everything is fine and working.

i.e we are throwing away signal that could make the system perform unexpected behaviour on another bad system.